text/profiles/0239-profiles-data-model.md

```diff
@@ -0,0 +1,1604 @@
+  11. A request is received, the system measures 1 request.
+  12. The 1 second collection cycle ends. A metric is exported for the
+      number of requests received over the interval of time t_1 to
+      t_0+1 with a value of 1.
+
+  Note: Even though, when reporting changes since last report time, using
+  CUMULATIVE is valid, it is not recommended. */
+  AGGREGATION_TEMPORALITY_CUMULATIVE = 2;
+}
+
+// ValueType describes the type and units of a value, with an optional aggregation temporality.
+message ValueType {
+  int64 type = 1; // Index into string table.
+  int64 unit = 2; // Index into string table.
+
+  AggregationTemporality aggregation_temporality = 3;
+}
+
+// Each Sample records values encountered in some program
+// context. The program context is typically a stack trace, perhaps
+// augmented with auxiliary information like the thread-id, some
+// indicator of a higher level request being handled etc.
+message Sample {
+  // The indices recorded here correspond to locations in Profile.location.
+  // The leaf is at location_index[0]. [deprecated, superseded by locations_start_index / locations_length]
+  repeated uint64 location_index = 1;
+  // locations_start_index along with locations_length refers to to a slice of locations in Profile.location.
+  // Supersedes location_index.
+  uint64 locations_start_index = 7;
+  // locations_length along with locations_start_index refers to a slice of locations in Profile.location.
+  // Supersedes location_index.
+  uint64 locations_length = 8;
+  // A 128bit id that uniquely identifies this stacktrace, globally. Index into string table. [optional]
+  uint32 stacktrace_id_index = 9;
+  // The type and unit of each value is defined by the corresponding
+  // entry in Profile.sample_type. All samples must have the same
+  // number of values, the same as the length of Profile.sample_type.
+  // When aggregating multiple samples into a single sample, the
+  // result has a list of values that is the element-wise sum of the
+  // lists of the originals.
+  repeated int64 value = 2;
+  // label includes additional context for this sample. It can include
+  // things like a thread id, allocation size, etc.
+  //
+  // NOTE: While possible, having multiple values for the same label key is
+  // strongly discouraged and should never be used. Most tools (e.g. pprof) do
+  // not have good (or any) support for multi-value labels. And an even more
+  // discouraged case is having a string label and a numeric label of the same
+  // name on a sample.  Again, possible to express, but should not be used.
+  // [deprecated, superseded by attributes]
+  repeated Label label = 3;
+  // References to attributes in Profile.attribute_table. [optional]
+  repeated uint64 attributes = 10;
+
+  // Reference to link in Profile.link_table. [optional]
+  uint64 link = 12;
+
+  // Timestamps associated with Sample represented in ms. These timestamps are expected
```

Comment:
In `ProfileContainer` there are `start_time_unix_nano` and `end_time_unix_nano`. Should we have the same precission with `timestamps` in `Sample` and also use ns instead of ms?

Reply:
With `Profile.time_nanos` there is another timestamp in the message that uses nanosecond precision.

Reply:
Discussed with profiling SiG.  We believe this discussion can continue as the model is implemented in the proto directory, and does not need to be addressed in the proposal, as it doesn't change overall direction.